src/txmempool.h

```diff
@@ -816,6 +790,102 @@ class CTxMemPool
         assert(m_epoch.guarded()); // verify guard even when it==nullopt
         return !it || visited(*it);
     }
+
+    /*
+     * CTxMemPoolChangeSet:
+     *
+     * This class is used for all mempool additions and associated removals (eg due to rbf).
```

Comment:
aside from reorgs?

Reply:
Reorgs are split: additions happen via the changesets (since we just use ATMP), while removals happen in their own special funky way.

I can update the comment.

Reply:
This is updated -- does this seem sufficient now, or should I add more documentation about how reorgs work?

Reply:
text is better thanks